lib/ohai/mixin/ec2_metadata.rb

```diff
@@ -200,6 +201,13 @@ def fetch_userdata
         response.code == "200" ? response.body : nil
       end
 
+      def fetch_dynamic_data
+        api_version = best_api_version
+        return nil if api_version.nil?
```

Comment:
this is going to trainwreck later with a "NoMethodError on NilClass" and should really handle this better